rtmp-client/src/main/java/net/butterflytv/rtmp_client/RtmpClient.java

```diff
@@ -16,44 +16,262 @@
+
+        /**
+         * Could not establish a socket connection to the server
+         */
+        public final static int SOCKET_CONNECT_FAIL = -9;
+
+        /**
+         * SOCKS negotiation failed
+         */
+        public final static int SOCKS_NEGOTIATION_FAIL = -10;
+
+        /**
+         * Could not create a socket to connect to RTMP server
+         */
+        public final static int SOCKET_CREATE_FAIL = -11;
+
+        /**
+         * SSL connection requested but not supported by the client
+         */
+        public final static int NO_SSL_TLS_SUPP = -12;
+
+        /**
+         * Could not connect to the server for handshake
+         */
+        public final static int HANDSHAKE_CONNECT_FAIL = -13;
+
+        /**
+         * Handshake with the server failed
+         */
+        public final static int HANDSHAKE_FAIL = -14;
+
+        /**
+         * RTMP server connection failed
+         */
+        public final static int RTMP_CONNECT_FAIL = -15;
+
+        /**
+         * Connection to the server lost
+         */
+        public final static int CONNECTION_LOST = -16;
+
+        /**
+         * Received an unexpected timestamp from the server
+         */
+        public final static int RTMP_KEYFRAME_TS_MISMATCH = -17;
 
+        /**
+         * The RTMP stream received is corrupted
+         */
+        public final static int RTMP_READ_CORRUPT_STREAM = -18;
+
+        /**
+         * Memory allocation failed
+         */
+        public final static int RTMP_MEM_ALLOC_FAIL = -19;
+
+        /**
+         * Stream indicated a bad datasize, could be corrupted
+         */
+        public final static int RTMP_STREAM_BAD_DATASIZE = -20;
+
+        /**
+         * RTMP packet received is too small
+         */
+        public final static int RTMP_PACKET_TOO_SMALL = -21;
+
+        /**
+         * Could not send packet to RTMP server
+         */
+        public final static int RTMP_SEND_PACKET_FAILED = -22;
+
+        /**
+         * Sending pause to the server failed
+         */
+        public final static int RTMP_PAUSE_FAIL = -23;
+
+        /**
+         * Missing a :// in the URL
+         */
+        public final static int URL_MISSING_PROTOCOL = -24;
+
+        /**
+         * Hostname is missing in the URL
+         */
+        public final static int URL_MISSING_HOSTNAME = -25;
+
+        /**
+         * The port number indicated in the URL is wrong.
+         */
+        public final static int URL_INCORRECT_PORT = -26;
+
+        /**
+         * The RTMP client is in an illegal state
+         */
+        public final static int RTMP_ILLEGAL_STATE = -27;
+
+        /**
+         * RTMP client has encountered an unexpected error
+         */
+        public final static int RTMP_GENERIC_ERROR = -28;
 
         public final int errorCode;
 
-        public RtmpIOException(int errorCode) {
+        public RtmpIOException(String message, int errorCode) {
+            super(message);
```

Comment:
I don't really see value in the messages. They're not useful for displaying to end users (because they don't get translated), and code/developers will be using the `errorCode` rather than the message to figure out what went wrong. Can we get rid of the messages and `createRtmpIOException`?

Reply:
The stack trace would then just say RtmpIOException is caught, without any error code or message. I noticed that the other exceptions (mostly IOExceptions) always have a message which give more information on the stack trace, so added this.

I could remove this if you see no value in it. Please let me know.

Reply:
As long as the stack trace is enough for a developer to work out what's wrong, I think that's enough. There's no need to make it super pretty for them. So I'd suggest setting the message to `"Rtmp error: " + errorCode` and leaving it there.

Reply:
Done